loopy/schedule/tree.py

```diff
@@ -0,0 +1,269 @@
+__copyright__ = "Copyright (C) 2022 Kaushik Kulkarni"
```

Comment:
It's hard for me to see here how this was changed vs. the original.

Reply:
I've split up the commits into Kaushik's original implementation ([7f50179](https://github.com/inducer/loopy/pull/694/commits/7f5017925112856784c8d0ff52942a3d0c8df616)), followed by my changes on top of that initial commit. Does that make the changes clearer?

Reply:
Thanks!